### PR TITLE
Set the @nearby_competitions object if creation fails.

### LINKED ISSUE #1016: undefined method `empty?' for nil:NilClass when editing a competition with validation errors
See:

image

@pedrosino, I believe this is due to the fact that [the validation error codepath](https://github.com/thewca/worldcubeassociation.org/blob/a8b10d4dbda23da6b70dac93a4bd9fa9830089da/WcaOnRails/app/controllers/competitions_controller.rb#L343-L351) causes us to render `edit` without first setting `@nearby_competitions`.

### PR DESCRIPTION
**Pull request by pedrosino**
Fixes #1016 (again)

### CODE REVIEW

File: WcaOnRails/spec/features/competition_management_spec.rb [hunk 1]
```diff
@@ -14,6 +14,22 @@
       click_button "Create Competition"
 
       expect(page).to have_text("Successfully created new competition!")
+
+      visit "/competitions/new"
+      click_button "Create Competition"
+
+      expect(page).to have_text("Name must end with a year")
```
Comment (viroulep) [inline WcaOnRails/spec/features/competition_management_spec.rb]: Very naive question here: shouldn't this be a separate scenario?
Like scenario 1 is creating a valid competition, scenario 2 is attempting to create an invalid competition?

File: WcaOnRails/spec/features/competition_management_spec.rb [hunk 2]
```diff
@@ -14,6 +14,22 @@
...
+    end
+
+    scenario "User clones an existing competition" do
```
Comment (viroulep) [inline WcaOnRails/spec/features/competition_management_spec.rb]: Maybe add in the description that you are testing the validation part? (like add "with validation errors" at the end)

File: WcaOnRails/spec/features/competition_management_spec.rb (hunk 1 above)
Comment (pedrosino) [inline WcaOnRails/spec/features/competition_management_spec.rb]: I guess you're right. I don't know much about this scenarios thing.

File: WcaOnRails/spec/features/competition_management_spec.rb (hunk 2 above)
Comment (pedrosino) [inline WcaOnRails/spec/features/competition_management_spec.rb]: Right.

**Comment by pedrosino** [comment]
>During the review I noticed that we have both `manage_competition_spec.rb` and `competition_management_spec.rb`, and there is at least one duplicate scenario in both file: is there a fundamental difference between them, or could we merge them up somehow?

I can't answer this, as I didn't create any of those files. I just added a test where it felt right :)

**Comment by pedrosino** [comment]
Pedro said
>I can't answer this, as I didn't create any of those files. I just added a test where it felt right :)

Actually I added a test to the file @jfly  told me to. Looks like he's responsible for the redundancy: https://github.com/thewca/worldcubeassociation.org/commit/ecc373a23b501e8742f5294ad67723a69596e79e 
(thanks @jonatanklosko for finding this)

**Comment by jfly** [comment]
> Actually I added a test to the file @jfly told me to. Looks like he's responsible for the redundancy: ecc373a
(thanks @jonatanklosko for finding this)

Lol! I did not realize we had both of those. @pedrosino, could you combine competition_management_spec.rb and manage_competition_spec.rb into one file? You can choose whichever name you prefer =)

**Comment by pedrosino** [comment]
I did some adjusting, but I'm not very sure if things look good.

**Comment by viroulep** [comment]
LGTM :)

**Comment by pedrosino** [comment]
I'm merging this to solve the issue. We can improve the tests later if something looks wrong.

**Pull request merged**